vignettes/reporting.rmd

```diff
@@ -0,0 +1,394 @@
+---
+title: "New reporting in teal"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 3 
+vignette: >
+  %\VignetteIndexEntry{New reporting in teal}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Introduction
+
+This document outlines a significant refactor of `teal.reporter` and accompanying enhancements in the `teal` framework for improved reporting functionalities.
+
+The primary goals of enhanced reporting were to:`
+
+*   Address limited customization capabilities of the previous `ReportCard` system, which was tightly integrated with `teal` module programming, making user modifications difficult.
+*   Resolve reproducibility challenges associated with the existing `teal.reporter` (see [teal.reporter issue #280](https://github.com/insightsengineering/teal.reporter/issues/280)).
+*   Incorporate solutions for enabling and disabling reporter functions globally and for single modules.
+*   Allow to customize report cards without modifying the `teal` module programming, thus enabling users to create their own report cards (and edit existing ones).
+*   Ensure that the new reporting system is lightweight and does not consume excessive disk space, especially for bookmarking purposes.
+*   Maintain or improve existing functionalities, including the ability to generate fully reproducible reports.
+*   Provide a more flexible and scalable design that can adapt to future needs (Editor module can be extended for new R classes).
+*   Export Editor functionality to allow users to edit report cards directly in the UI.
+*   Introduce a new feature to disable the reporter functionality for one or multiple `teal` modules. 
+*   Improve the user experience by providing a more intuitive interface for report generation and customization (changes in Reporter Previewer UI).
+*   Ensure that the new system is compatible with existing `teal` modules and does not break any existing functionality.
+
+To summarize, the refactor aimed for a more flexible and scalable design, ensuring that the new system supports bookmarking without excessive disk space, allows report card customization without altering `teal` module programming, 
+generates fully reproducible reports, and maintains or improves existing functionalities. Additionally, a key new feature is the ability for users to disable the reporter functionality for one or multiple `teal` modules.
+
+## Key Changes and New Features
```

Comment:
How about a small summary of all 3 here and then keep the sections as is?

Reply:
Basically, add something like below:

- Introduction of `ReportDocument`: Replaces the R6-based `ReportCard` with a lightweight, flexible S3 list structure that simplifies report creation, reduces storage size, and supports easy content editing and reordering.
- Enhanced Integration with `teal`: New capabilities include global/module-level control over reporting, support for custom templates, and programmatic report output modification via `modify_reactive_output()` for greater flexibility.
- Improved Reproducibility and Code Display: Refined download/upload mechanisms ensure full report reproducibility, while updates to `teal.code` enhance the clarity and labeling of code in generated reports.